docs/api/js/utils.md

```diff
@@ -1,68 +1,563 @@
+
+```ts
+async function isEIP1271SignatureCorrect(
+    provider: Provider,
+    address: string,
+    msgHash: string,
+    signature: SignatureLike
+): Promise<boolean> {
+    const accountContract = new ethers.Contract(address, IERC1271, provider);
+
+    // This line may throw an exception if the contract does not implement the EIP1271 correctly.
+    // But it may also throw an exception in case the internet connection is lost.
+    // It is the caller's responsibility to handle the exception.
+    const result = await accountContract.isValidSignature(msgHash, signature);
+
+    return result == EIP1271_MAGIC_VALUE;
+}
+```
+
+### `isMessageSignatureCorrect`
+
+Returns true if account abstraction EIP712 signature is correct.
+
+```ts
+export async function isMessageSignatureCorrect(
+    provider: Provider,
+    address: string,
+    message: ethers.Bytes | string,
+    signature: SignatureLike
+): Promise<boolean> {
+    const msgHash = ethers.utils.hashMessage(message);
+    return await isSignatureCorrect(provider, address, msgHash, signature);
+}
+```
+
+
+### `isSignatureCorrect`
+
+Called from [`isMessageSignatureCorrect`](#ismessagesignaturecorrect) and [`isTypedDataSignatureCorrect`](#istypeddatasignaturecorrect). Returns true if account abstraction EIP712 signature is correct.
+
+```ts
+async function isSignatureCorrect(
+    provider: Provider,
+    address: string,
+    msgHash: string,
+    signature: SignatureLike
+): Promise<boolean> {
+    let isContractAccount = false;
+
+    const code = await provider.getCode(address);
+    isContractAccount = ethers.utils.arrayify(code).length != 0;
+
+    if (!isContractAccount) {
+        return isECDSASignatureCorrect(address, msgHash, signature);
+    } else {
+        return await isEIP1271SignatureCorrect(provider, address, msgHash, signature);
+    }
+}
+```
+
+### `isTypedDataSignatureCorrect`
+
+Returns true if account abstraction EIP712 signature is correct.
+
+```ts
+export async function isTypedDataSignatureCorrect(
+    provider: Provider,
+    address: string,
+    domain: TypedDataDomain,
+    types: Record<string, Array<TypedDataField>>,
+    value: Record<string, any>,
+    signature: SignatureLike
+): Promise<boolean> {
+    const msgHash = ethers.utils._TypedDataEncoder.hash(domain, types, value);
+    return await isSignatureCorrect(provider, address, msgHash, signature);
+}
+```
+
+### `parseTransaction`
+
+Common parsing transaction function used by internal teams. 
+
+Please see the [utilities library definition](https://github.com/matter-labs/zksync-2-dev/blob/94701bd2fbc590f733346934cfbccae08fc62f1a/sdk/zksync-web3.js/src/utils.ts) for more info.
+
+### `serialize`
+
+Common serialize function used by internal teams. 
+
+Please see the [utilities library definition](https://github.com/matter-labs/zksync-2-dev/blob/94701bd2fbc590f733346934cfbccae08fc62f1a/sdk/zksync-web3.js/src/utils.ts) for more info.
+
+### `serializeTransaction`
```

Comment:
this is ethers function`node_modules/@ethersproject/transactions/src.ts/index.ts`

Reply:
we do not need to mention it here

Reply:
Removed serializeTransaction function